initializr-generator-spring/src/main/java/io/spring/initializr/generator/spring/scm/git/GitEnum.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.spring.initializr.generator.spring.scm.git;
+
+/**
+ * Default configurations for gitignore file.
+ * @author Yves Galvão
+ */
+public enum GitEnum {
```

Comment:
I am not sure why we'd want to move this to an enum. Can you expand why you think this improves the status quo?

Reply:
Hi @snicoll, I made some customs in the code for the company where I work, and looking for SOLID principles, I thinking that's flow is violated the [Open-closed principle](https://en.wikipedia.org/wiki/Open–closed_principle). If another **GitIgnoreSection** is needed on the flow, the developer should by modify the **GitIgnore class**, the **write method on GitIgnore class**, the **createGitIgnore method on GitProjectGenerationConfiguration**, and **add a custom method to return the list of ignores on the GitIgnore class**. In contrast, with this refactoring the only one adjustment is to **add a new value on the created GitEnum**.